dockerfiles/theia/Dockerfile

```diff
@@ -23,16 +24,18 @@ WORKDIR /home/theia
 ADD https://raw.githubusercontent.com/theia-ide/theia/v${THEIA_VERSION}/examples/browser/package.json /home/theia/package.json
 ADD theia-default-package.json /home/default/theia/package.json
 ADD versions.sh /home/default/versions.sh
-RUN cd /home/default && ./versions.sh
-
 ADD supervisord.conf /etc/
-RUN cd /home/theia && \
+RUN cd /home/default && ./versions.sh && \
+    cd /home/theia && \
     yarn && \
     yarn theia build --mode development && \
     rm -rf * && \
     cd /home/default/theia && \
     yarn && \
     yarn theia build
+RUN find ${YARN_CACHE_FOLDER} -exec sh -c "chgrp 0 {}; chmod g+rwX {}" \;
```

Comment:
should we group all RUN instructions together ?

Reply:
Do not think so. It could be helpful if we add something and then delete it at the end of `RUN`. But in case it still remains we cannot reduce image size. What we will have - a single layer with all the changes, which is not so good, because it makes using caches harder.

Reply:
But it will happen anyway and doesn't depend on number of layers...